src/prefect/cli/deploy.py

```diff
@@ -600,7 +601,7 @@ async def _run_single_deploy(
         triggers = []
 
     pull_steps = (
-        deploy_config.get("pull")
+        pull_steps
         or actions.get("pull")
```

Comment:
would we still need this line? it looks like thats now on 340 as well

Reply:
nevermind, we may have modified `actions`

Reply:
Yeah, there are some changes to `actions` that happen when we generate a pull step that need to be available here.